addons/zeus/functions/fnc_ui_spectator.sqf

```diff
@@ -0,0 +1,265 @@
+    };
+
+    _ctrl ctrlSetTextColor _color;
+    [_ctrl, _scale, 0.1] call BIS_fnc_ctrlSetScale;
+};
+
+// Use the unit's side as default
+private _activeSide = [east, west, independent, civilian] find _side;
+
+// Handle sides other than default four (sideEnemy)
+if (_activeSide != -1) then {
+    SETVAR(_display,spectateSides,[_activeSide]);
+};
+
+{
+    private _ctrl = _display displayCtrl _x;
+    private _side = _x - 92540;
+    private _color = [_side] call BIS_fnc_sideColor;
+    _ctrl setVariable ["color", _color];
+    _ctrl ctrlSetActiveColor _color;
+    _color set [3, 0.5];
+
+    if (_side == _activeSide) then {
+        [_ctrl, 1.2, 0] call BIS_fnc_ctrlSetScale;
+        _color set [3, 1];
+    };
+
+    _ctrl ctrlSetTextColor _color;
+
+    _ctrl ctrlAddEventHandler ["ButtonClick", _fnc_onSideSelection];
+} forEach SIDE_IDCs;
+
+// Camera modes
+private _fnc_onModesSelection = {
+    params ["_ctrl"];
+
+    private _display = ctrlParent _ctrl;
+    if (isNull _display) exitWith {};
+
+    private _color = [1, 1, 1, 0.5];
+    private _scale = 1;
+
+    private _modes = GETVAR(_display,cameraModes,[]);
+    private _selectedMode = (ctrlIDC _ctrl) - 92550;
+
+    // Add or remove from camera modes and update color and scale
+    if (_selectedMode in _modes) then {
+        SETVAR(_display,cameraModes,_modes - [_selectedMode]);
+    } else {
+        SETVAR(_display,cameraModes,_modes + [_selectedMode]);
+        _color set [3, 1];
+        _scale = 1.2;
+    };
+
+    _ctrl ctrlSetTextColor _color;
+    [_ctrl, _scale, 0.1] call BIS_fnc_ctrlSetScale;
+};
+
+// Use setting as default since global variable will change
+private _availableModes = [[0,1,2], [1,2], [0], [1], [2]] select EGVAR(spectator,restrictModes);
+SETVAR(_display,cameraModes,_availableModes);
+
+{
+    private _ctrl = _display displayCtrl _x;
+    private _color = [1, 1, 1, 0.5];
+
+    if ((_x - 92550) in _availableModes) then {
+        [_ctrl, 1.2, 0] call BIS_fnc_ctrlSetScale;
+        _color set [3, 1];
+    };
+
+    _ctrl ctrlSetTextColor _color;
+
+    _ctrl ctrlAddEventHandler ["ButtonClick", _fnc_onModesSelection];
+} forEach CAMERA_IDCs;
+
+// Vision Modes
+private _fnc_onVisionSelection = {
+    params ["_ctrl", "_state"];
+
+    private _display = ctrlParent _ctrl;
+    if (isNull _display) exitwith {};
+
+    // Convert to boolean since EH returns state as 0 or 1
+    private _state = [false, true] select _state;
+
+    private _visions = GETVAR(_display,visionModes,[]);
+    private _selectedVision = (ctrlIDC _ctrl) - 92560;
+
+    // Add or remove from vision modes
+    if (_state) then {
+        SETVAR(_display,visionModes,_visions + [_selectedVision]);
+    } else {
+        SETVAR(_display,visionModes,_visions - [_selectedVision]);
+    };
+
+    // Handle all checked/unchecked
+    private _allCheckboxes = VISION_IDCs apply {cbChecked (_display displayCtrl _x)};
+
+    if (({_x isEqualTo _state} count _allCheckboxes) isEqualTo 4) then {
```

Comment:
Not sure if best way to check if all checkboxes are checked or unchecked. Could also do:
`count (_allCheckboxes arrayIntersect _allCheckboxes) == 1`